test/gpu/native/threadBlockAndGridPrimitives.chpl

```diff
@@ -105,7 +105,8 @@ proc runExample(gdimX, gdimY, gdimZ, bdimX, bdimY, bdimZ) {
   writeln("Block size: ", bdimX, " x ", bdimY, " x ", bdimZ);
 
   var deviceBuffer = getDeviceBufferPointer(gdimX, gdimY, gdimZ, bdimX, bdimY, bdimZ);
-  __primitive("gpu kernel launch", c"add_nums",
+  __primitive("gpu kernel launch",
+              c".threadBlockAndGridPrimitives_files/chpl__gpu.fatbin", c"add_nums",
```

Comment:
Yeah I think we should just fold `gpu kernel launch` and `gpu kernel launch flat`. 

If we're convinced that we'll only have flat kernel launches I think we can just get rid of the extra code we're adding to handle 3d blocks\grids and remove any tests that rely on it.

If we're unsure than maybe we could make the function polymoprhic: you can either pass in the block\grid size as a single number or a 2 or 3-tuple of dims.

Not something I want to worry about for this PR though.

Reply:
> If we're convinced that we'll only have flat kernel launches I think we can just get rid of the extra code we're adding to handle 3d blocks\grids and remove any tests that rely on it.

I still don't want to say that we'll never have 3D kernel launches. This is partly because of https://github.com/Cray/chapel-private/issues/2357.

That being said, with this fatbin argument dropped from both of the primitives, I agree that this is not a discussion for this PR.